File: PetCafeProject/Models/Venda.cs
using System.ComponentModel.DataAnnotations;

namespace PetCafeProject.Models
{
    public class Venda
    {
        [Key]
        public int Id { get; set; } 

        public Cliente Cliente { get; set; }
        public Produto Produto { get; set; }
        public int Quantidade { get; set; }
        public double ValorVenda { get; set; }

        public Venda()
        {
        }

        public Venda(Cliente cliente, Produto produto, int quantidade, double valorVenda)
        {
            Cliente = cliente;
            Produto = produto;
            Quantidade = quantidade;
            ValorVenda = valorVenda;
        }
    }
}

File: AppPetCafe/src/app/Model/Venda.ts
import { Cliente } from "./Cliente";
import { Produto } from "./Produto";

export class Venda {

    id: number = 0;
    clienteCPF: string = '';
    cliente: Cliente | undefined;
    produtoCodigo: number = 0;
    protudo: Produto | undefined;
    quantidade: number = 0;
    valorVenda: number = 0;
  produto: any;
}